text_extensions_for_pandas/io/conll.py

```diff
@@ -462,6 +462,46 @@ def iob_to_spans(
         )
 
 
+def spans_to_iob(
+    span_frame: pd.DataFrame,
+    token_span_col_name: str = "token_span"
+):
+    """
+    Convert a series of `TokenSpan`s of entities to token tags in
+    Inside–Outside–Beginning (IOB) format.
+    :param span_frame: DataFrame with a `TokenSpanArray` column.
+    :param token_span_col_name: Name of a column in `token_features` that contains the
+     IOB tags as strings, "I", "O", or "B".
+    :return: A `pd.DataFrame` with the following columns:
+    * `ent_iob`: entity type in IOB format.
+    """
+    token_spans = span_frame[token_span_col_name]
+
+    # Define the IOB categorical type with 'B'==0, 'I'==1, 'O'==2
+    dtype = pd.CategoricalDtype(['B', 'I', 'O'], ordered=False)
+
+    # Handle an empty token span array
+    if len(token_spans) == 0:
+        return pd.DataFrame({'ent_iob': pd.Categorical([], dtype)})
+
+    # Initialize an IOB series with all 'O' entities
+    iob_data = np.ones(len(token_spans[0].tokens), dtype='int64') * 2
```

Comment:
If you use zero for `O`, then `np.ones(custom shape) * 2` becomes `np.zeros_like` here

Reply:
I was trying to follow the category codes that I saw somewhere else, which looked like 'O' == 2. Would it help to define these categories as a list somewhere?

Reply:
You probably saw Pandas assigning arbitrary numbers to the distinct values. If you're coding the I, O, and B manually, then it's better for O to be zero because that tag means "zero entities present at this position".